clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -666,7 +686,8 @@ protected Map<String, Object> postProcessParsedConfig(final Map<String, Object>
         Map<String, Object> refinedConfigs = CommonClientConfigs.postProcessReconnectBackoffConfigs(this, parsedValues);
         maybeOverrideClientId(refinedConfigs);
         maybeOverrideEnableAutoCommit(refinedConfigs);
-        checkGroupRemoteAssignor();
+        checkUnsupportedConfigs(GroupProtocol.CLASSIC, CLASSIC_PROTOCOL_UNSUPPORTED_CONFIGS);
+        checkUnsupportedConfigs(GroupProtocol.CONSUMER, CONSUMER_PROTOCOL_UNSUPPORTED_CONFIGS);
```

Comment:
I prefer the current approach because it allows us to reuse this method in the future if new protocols are added, without having to add additional if-else statements to the original method. WDYT?

Reply:
my reasoning was that consumer group protocols is not something that I expect we'll be adding frequently, and seemed a bit confusing to check for the 2 protocols, when a consumer can only be using one (that being said, no strong position here, I see your point of view too, so up to you)

Reply:
I think that we won't add new protocol frequently, so it make snese to me, I will change it to `unsupportedConfigs`.